File: Services/Orders/Model/Order.cs
using System.Globalization;
using System.Text.Json.Serialization;

namespace Orders.Model
{
    public class Order
    {   
        //[JsonIgnore]
        public int Id { get; set; }

        public string OwnerGuid { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public OrderKind Kind { get; set; }
        
        public int Count { get; set; }
        
        public decimal Price { get; set; }
        
        [JsonConverter(typeof(JsonStringEnumConverter))]           
        public OrdersCurrency BaseCurrency { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public OrdersCurrency QuoteCurrency { get; set; }

        public Order()
        {
            
        }

        public Order(int id, 
                     string ownerguid, 
                     OrderKind kind,
                     int count, 
                     decimal price, 
                     OrdersCurrency basecurrency, 
                     OrdersCurrency quotecurrency)
        {
            this.Id = id;
            this.OwnerGuid = ownerguid;
            this.Kind = kind;
            this.Count = count;
            this.Price = price;
            this.BaseCurrency = basecurrency;
            this.QuoteCurrency = quotecurrency;
        }

        public Order(string ownerguid, 
                     OrderKind kind,
                     int count, 
                     decimal price, 
                     OrdersCurrency basecurrency, 
                     OrdersCurrency quotecurrency)
        {
            this.OwnerGuid = ownerguid;
            this.Kind = kind;
            this.Count = count;
            this.Price = price;
            this.BaseCurrency = basecurrency;
            this.QuoteCurrency = quotecurrency;
        }
    }
}

File: Services/Portfolio/models/order.ts
import * as Sequelize from 'sequelize';
import { DataTypes, Model, Optional } from 'sequelize';

export interface OrderAttributes {
  id: number;
  ownerguid: string;
  kind?: "Buy" | "Sell";
  count?: number;
  price?: number;
  basecurrency?: "BTC" | "ETH" | "DASH";
  quotecurrency?: "BTC" | "ETH" | "DASH";
}

export type OrderPk = "id";
export type OrderId = Order[OrderPk];
export type OrderOptionalAttributes = "id" | "kind" | "count" | "price" | "basecurrency" | "quotecurrency";
export type OrderCreationAttributes = Optional<OrderAttributes, OrderOptionalAttributes>;

export class Order extends Model<OrderAttributes, OrderCreationAttributes> implements OrderAttributes {
  id!: number;
  ownerguid!: string;
  kind?: "Buy" | "Sell";
  count?: number;
  price?: number;
  basecurrency?: "BTC" | "ETH" | "DASH";
  quotecurrency?: "BTC" | "ETH" | "DASH";


  static initModel(sequelize: Sequelize.Sequelize): typeof Order {
    return Order.init({
    id: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    ownerguid: {
      type: DataTypes.STRING(50),
      allowNull: false
    },
    kind: {
      type: DataTypes.ENUM("Buy","Sell"),
      allowNull: true
    },
    count: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    price: {
      type: DataTypes.DECIMAL,
      allowNull: true
    },
    basecurrency: {
      type: DataTypes.ENUM("BTC","ETH","DASH"),
      allowNull: true
    },
    quotecurrency: {
      type: DataTypes.ENUM("BTC","ETH","DASH"),
      allowNull: true
    }
  }, {
    sequelize,
    tableName: 'orders',
    schema: 'public',
    timestamps: false,
    indexes: [
      {
        name: "orders_pkey",
        unique: true,
        fields: [
          { name: "id" },
        ]
      },
    ]
  });
  }
}
